File: python/azure-ai-multi-agent/ai/agent/common/chat_response_extensions.py
# ---------------------------------------------------------
# Copyright (c) Microsoft Corporation. All rights reserved.
# ---------------------------------------------------------

from typing import List
from .chat_response_update import ChatResponseUpdate
from .chat_response import ChatResponse
from .chat_message import ChatMessage

def chat_response_updates_to_chat_response(updates: List[ChatResponseUpdate]) -> ChatResponse:
    """
    Converts a list of ChatResponseUpdate to a ChatResponse.
    
    Args:
        updates (List[ChatResponseUpdate]): The list of response updates to convert.
        
    Returns:
        ChatResponse: The converted chat response.
    """
    response = ChatResponse()
    
    # Create a new ChatMessage from the updates
    if updates:
        message = ChatMessage(updates[0].role, "")
        message.author_name = updates[0].author_name
        
        for update in updates:
            if update.content:
                message.content += update.content
                
        response.messages.append(message)
    
    return response

File: dotnet/src/Microsoft.Extensions.AI.Agents/ChatCompletion/ChatResponseExtensions.cs
// Copyright (c) Microsoft. All rights reserved.

using Microsoft.Shared.Diagnostics;

namespace Microsoft.Extensions.AI.Agents;

/// <summary>
/// Contains extension methods for <see cref="ChatResponse"/> and <see cref="ChatResponseUpdate"/>.
/// </summary>
internal static class ChatResponseExtensions
{
    /// <summary>
    /// Converts a <see cref="ChatResponse"/> instance to an <see cref="AgentRunResponse"/>.
    /// </summary>
    /// <param name="chatResponse">The <see cref="ChatResponse"/> to convert. Cannot be <see langword="null"/>.</param>
    /// <returns>
    /// An <see cref="AgentRunResponse"/> containing the messages, metadata, and additional properties from the
    /// specified <see cref="ChatResponse"/>.
    /// </returns>
    public static AgentRunResponse ToAgentRunResponse(this ChatResponse chatResponse)
    {
        _ = Throw.IfNull(chatResponse);

        return new AgentRunResponse(chatResponse.Messages)
        {
            ResponseId = chatResponse.ResponseId,
            CreatedAt = chatResponse.CreatedAt,
            Usage = chatResponse.Usage,
            RawRepresentation = chatResponse,
            AdditionalProperties = chatResponse.AdditionalProperties
        };
    }

    /// <summary>
    /// Converts a <see cref="ChatResponseUpdate"/> instance to an <see cref="AgentRunResponseUpdate"/>.
    /// </summary>
    /// <param name="chatResponseUpdate">The <see cref="ChatResponseUpdate"/> to convert. Cannot be <see langword="null"/>.</param>
    /// <returns>An <see cref="AgentRunResponseUpdate"/> containing the properties from the specified <see cref="ChatResponseUpdate"/>.</returns>
    public static AgentRunResponseUpdate ToAgentRunResponseUpdate(this ChatResponseUpdate chatResponseUpdate)
    {
        _ = Throw.IfNull(chatResponseUpdate);

        return new AgentRunResponseUpdate()
        {
            Role = chatResponseUpdate.Role,
            AuthorName = chatResponseUpdate.AuthorName,
            Contents = chatResponseUpdate.Contents,
            MessageId = chatResponseUpdate.MessageId,
            ResponseId = chatResponseUpdate.ResponseId,
            CreatedAt = chatResponseUpdate.CreatedAt,
            RawRepresentation = chatResponseUpdate,
            AdditionalProperties = chatResponseUpdate.AdditionalProperties
        };
    }
}
